Datamine/Modules/SpellData.lua

```diff
@@ -3,9 +3,15 @@ Datamine.Spell = CreateFrame("Frame");
 local L = Datamine.Strings;
 local moduleName = L.SPELL_INFO_MODULE_NAME;
 
-local GetSpellInfo = C_Spell and C_Spell.GetSpellInfo or GetSpellInfo;
+local GetSpellInfo = GetSpellInfo or function(spell)
+    local spellInfo = C_Spell.GetSpellInfo(spell);
+    if spellInfo then
+        return spellInfo.name, 0, spellInfo.iconID, spellInfo.castTime, spellInfo.minRange, spellInfo.maxRange, spellInfo.spellID, spellInfo.originalIconID;
+    end
+    return nil;
+end
 local GetSpellLink = C_Spell and C_Spell.GetSpellLink or GetSpellLink;
-local GetSpellDescription = C_Spell and C_Spell.GetSpellDescription or GetSpellDescription;
+local GetSpellDescription = GetSpellDescription;
```

Comment:
why not just delete this line entirely

Reply:
Nothin wrong with a bit of localized functions for caching :D

Reply:
Taking things as upvalues is not a free and guaranteed performance boost, and is broadly unnecessary.

Performance guides should generally always come with a "profile before you optimize" note, but I understand that for most people reality is boring and it's easier to just follow the bulletpoints and pretend that they're relevant for the situation at hand.

Upvalues are heap allocated as their own individual objects. Each unique allocation for an upvalue is not guaranteed or even consistently likely to have any semblance of locality with any others allocated around the same time, nor any locality to the function prototype object or the closure object. As such, one should expect that _infrequent_ access to upvalues are likely to result in cache misses.

Contrast this with tables; tables pack hash nodes together densely in a single contiguous memory block, and even hash collisions are resolved by chaining within that same block without requiring further heap allocations. For commonly accessed tables like `_G`, you'd likely find that on any given game tick there's a decent chunk of that block sitting resident in the cache, especially if Blizzard or other addons are calling the same functions that you are.

In a hot loop of 1e10 iterations upvalues will look great because you've got an access pattern that's perfect for them - one pointer deference and it's done. The problem is most of your code is not going to be a hot loop, and in those cases you should expect mediocrity from this eager and unprofiled optimization.